monogen/monogen.go

```diff
@@ -0,0 +1,263 @@
+package monogen
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/gobuffalo/genny/v2"
+	"github.com/ignite/cli/v28/ignite/config"
+	"github.com/ignite/cli/v28/ignite/pkg/cache"
+	"github.com/ignite/cli/v28/ignite/pkg/gocmd"
+	"github.com/ignite/cli/v28/ignite/pkg/placeholder"
+	"github.com/ignite/cli/v28/ignite/pkg/xast"
+	"github.com/ignite/cli/v28/ignite/services/scaffolder"
+	"github.com/ignite/cli/v28/ignite/templates/module"
+)
+
+func Generate(ctx context.Context, appDirPath, goModulePath, addressPrefix string, skipGit bool) error {
+	// TODO: don't do anything if directory already exists. I feel like ignite should handle this somewhere.
+
+	igniteRootDir, err := config.DirPath()
+	if err != nil {
+		return fmt.Errorf("get ignite config directory: %v", err)
+	}
+	// https://github.com/ignite/cli/blob/2a968e8684cae0a1d79ccb11f0db067a4605173e/ignite/cmd/cmd.go#L33-L34
+	cacheDBPath := filepath.Join(igniteRootDir, "ignite_cache.db")
+	// Note that we need to change this version every time we bump ignite's version in Monomer's go.mod.
+	cacheStorage, err := cache.NewStorage(cacheDBPath, cache.WithVersion("v28.5.1"))
```

Comment:
Would it make sense to just read the ignite version directly from `go.mod` to avoid needing to change this manually if the ignite version is ever updated?

Reply:
We would need the path to our own `go.mod`, which may not even be present on a user's machine.

However, we might be able to embed the `go.mod` in the `monogen.go` file and read it directly...

This is actually a really interesting idea. Will try.

Reply:
I forgot we can't `go:embed` files from parent directories, so we'd need to either make monogen a separate module or copy the `go.mod` file to the `monogen` directory. Neither option is super appealing to me.

I decided to use the `version.Version` variable from ignite, which is set to `development` when accessed programatically. When the ignite binary is built, this variable is set to the release tag, so we won't be able to use the same cache, but at least we shouldn't corrupt anything.